homeassistant/components/evohome.py

```diff
@@ -0,0 +1,690 @@
+                "Failed to establish a connection with evohome web servers, "
+                "Check your username (%s), and password are correct."
+                "Unable to continue. Resolve any errors and restart HA.",
+                domain_data['params'][CONF_USERNAME]
+            )
+        else:
+            # back off and try again later
+            raise PlatformNotReady(err)
+
+    finally:  # redact username, password as no longer needed
+        del domain_data['params'][CONF_USERNAME]  # = 'REDACTED'
+        del domain_data['params'][CONF_PASSWORD]  # = 'REDACTED'
+
+    domain_data['client'] = client
+
+
+# 3. REDACT any installation data we'll never need
+    if client.installation_info[0]['locationInfo']['locationId'] != 'REDACTED':
+        for loc in client.installation_info:
+            loc['locationInfo']['locationId'] = 'REDACTED'
+            loc['locationInfo']['streetAddress'] = 'REDACTED'
+            loc['locationInfo']['city'] = 'REDACTED'
+            loc['locationInfo']['locationOwner'] = 'REDACTED'
+            loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+
+# 3a.
+    loc_idx = domain_data['params'][CONF_LOCATION_IDX]
+
+    try:
+        domain_data['config'] = client.installation_info[loc_idx]
+
+    # IndexError: configured location index is outside the range this login
+    except IndexError:
+        _LOGGER.warning(
+            "setup(): Config parameter, '%s'= %s , is out of range (0-%s), "
+            "continuing with '%s' = 0.",
+            CONF_LOCATION_IDX,
+            loc_idx,
+            len(client.installation_info) - 1,
+            CONF_LOCATION_IDX
+        )
+
+        domain_data['params'][CONF_LOCATION_IDX] = 0
+        domain_data['config'] = client.installation_info[0]
+
+    domain_data['status'] = {}
+
+# 3. Finished - do we need to output debgging info? If so...
+    if _LOGGER.isEnabledFor(logging.DEBUG):
+        _LOGGER.debug(
+            "setup(): Location/TCS (temp. control system) used is: %s [%s]",
+            domain_data['config'][GWS][0][TCS][0]['systemId'],
+            domain_data['config']['locationInfo']['name']
+        )
+        # Some of this data needs further redaction before being logged
+        tmp = dict(domain_data['config'])
+        tmp['locationInfo']['postcode'] = 'REDACTED'
+
+        _LOGGER.debug("setup(): domain_data['config']: %s", tmp)
+
+
+# Now we're ready to go, but we have no state as yet, so...
+    def _first_update(event):
+        """Let the controller know it can obtain it's first update."""
+    # send a message to the master to do its first update
+        pkt = {
+            'sender': 'setup()',
+            'signal': 'update',
+            'to': EVO_MASTER
+        }
+        hass.helpers.dispatcher.async_dispatcher_send(
+            DISPATCHER_EVOHOME,
+            pkt
+        )
+
+# create a listener for the above...
+    hass.bus.listen(EVENT_HOMEASSISTANT_START, _first_update)
+
+# ... then finally, load the platform...
```

Comment:
Again, too many comments. Do we really need a comment stating "load the platform" before calling the `load_platorm` function?

Reply:
removed